tests/test_mongodb_log_based_interruptible.py

```diff
@@ -0,0 +1,330 @@
+            interrupted_state['bookmarks']['simple_db-'+table_interrupted].update({'last_id_fetched_type': 'ObjectId'})
+            interrupted_state['bookmarks']['simple_db-'+table_interrupted].update(last_id)
+            interrupted_state['currently_syncing'] = 'simple_db-'+table_interrupted
+            versions[tap_stream_id] = version
+
+            # Insert Update Delete few records in the collection
+
+            doc_to_update_1 = client["simple_db"]["simple_coll_1"].find_one()
+            client["simple_db"]["simple_coll_1"].find_one_and_update({"_id": doc_to_update_1["_id"]}, {"$set": {"int_field": 999}})
+
+            doc_to_update_2 = client["simple_db"]["simple_coll_2"].find_one()
+            client["simple_db"]["simple_coll_2"].find_one_and_update({"_id": doc_to_update_2["_id"]}, {"$set": {"int_field": 888}})
+
+            doc_to_delete_1 = client["simple_db"]["simple_coll_1"].find_one({"int_field": 2})
+            client["simple_db"]["simple_coll_1"].delete_one({"_id": doc_to_delete_1["_id"]})
+
+            doc_to_delete_2 = client["simple_db"]["simple_coll_2"].find_one({"int_field": 2})
+            client["simple_db"]["simple_coll_2"].delete_one({"_id": doc_to_delete_2["_id"]})
+
+            last_inserted_coll_1 = client["simple_db"]["simple_coll_1"].insert_many(generate_simple_coll_docs(1))
+            # get the last inserted id in coll 1
+            last_inserted_id_coll_1 = str(last_inserted_coll_1.inserted_ids[0])
+
+            last_inserted_coll_2 = client["simple_db"]["simple_coll_2"].insert_many(generate_simple_coll_docs(1))
+            # get the last inserted id in coll 2
+            last_inserted_id_coll_2 = str(last_inserted_coll_2.inserted_ids[0])
+
+        menagerie.set_state(conn_id, interrupted_state)
+
+        ### Run 2nd sync ###
+
+        second_sync = runner.run_sync_mode(self, conn_id)
+        second_sync_exit_status = menagerie.get_exit_status(conn_id, second_sync)
+        menagerie.verify_sync_exit_status(self, second_sync_exit_status, second_sync)
+
+        records_by_stream_2 = runner.get_records_from_target_output()
+
+        record_count_by_stream_2 = runner.examine_target_output_file(self,
+                                                                     conn_id,
+                                                                     self.expected_sync_streams(),
+                                                                     self.expected_pks())
+
+        # Verify the record count for the 2nd sync
+        for tap_stream_id in self.expected_sync_streams():
+            self.assertGreaterEqual(record_count_by_stream_2[tap_stream_id], self.expected_row_count_2()[tap_stream_id])
+
+        second_state = menagerie.get_state(conn_id)
+
+        # Verify the interrupted table is synced first
+        second_sync_records = list(records_by_stream_2)
+        self.assertEqual(second_sync_records[0], table_interrupted)
```

Comment:
I am not sure if this actually verifies the order that the tables are synced. The `records_by_stream_2` is a dictionary, which should maintain ordering of keys since we are running 3.8 and 3.9 for tap-tester in most places now. But I think that ordering is set by insertions to the dictionary. And the insertions are made by parsing the target-output file line by line which is just a giant json blob. At the top level it is an object which is not ordered. The messages value for each table is an array which is ordered. But just grabbing the first key from this dictionary, based on the first key value pair read from a json object, does not necessarily reflect the first stream that was replicatied.

Reply:
removing assertion as per our discussion